### ISSUE TITLE
DC1 VPN access

### ISSUE THREAD

**Issue by helen-d-little** [alan-turing-institute/data-safe-haven#2299]
## :white_check_mark: Checklist

- [x] I have searched open and closed issues for duplicates.
- [x] This is a problem observed when **managing** a Data Safe Haven.
- [ ] I can reproduce this with the [latest version](https://github.com/alan-turing-institute/data-safe-haven/releases).
- [x] I have read through the [documentation](https://alan-turing-institute.github.io/data-safe-haven/).
- [x] This isn't an open-ended question (open a [discussion](https://github.com/alan-turing-institute/data-safe-haven/discussions) if it is).

## :computer: System information

- **Operating System:** 
- **Data Safe Haven version:** 

## :package: Packages

<details>
<summary>List of packages</summary>

```none
Paste list of packages here
```

</details>

## :no_entry_sign: Describe the problem
Can no longer connect to the VPN for DC1 access for Prod4 ([docs 4.2.2 link](https://data-safe-haven.readthedocs.io/en/v4.2.2/deployment/deploy_shm.html#configure-vpn-connection)) as the client certificate expired on 31/10/2024
## :steam_locomotive: Workarounds or solutions
create a new self-signed certificate with the correct options (see Setup_SHM_Key_Vault_And_Emergency_Admin.ps1), replace the certificate in the keyvault then update the vpn gateway

**Comment by JimMadge** [comment]
Just to confirm, you were able to generate a new cert and use that?

This infrastructure isn't present in the latest versions so I think there is nothing to fix here.

**Comment by jemrobinson** [comment]
There isn't a standalone script to generate a new certificate. Minimal solution using existing scripts would be to:

- delete the expired certificate in the key vault
- delete the VPN gateway
- run `./Setup_SHM_Key_Vault_And_Emergency_Admin.ps1` to generate a new certificate
- run `./Setup_SHM_Networking.ps1` to deploy a new VPN gateway with the certificate

Minimal solution without using existing scripts is as @helendduncan suggests above.

**Comment by helen-d-little** [comment]
> There isn't a standalone script to generate a new certificate. Minimal solution using existing scripts would be to:
> 
>  - [x] delete the expired certificate in the key vault
> - [x] delete the VPN gateway
> - [x] run `./Setup_SHM_Key_Vault_And_Emergency_Admin.ps1` to generate a new certificate
> - [ ] run `./Setup_SHM_Networking.ps1` to deploy a new VPN gateway with the certificate
> 
> Minimal solution without using existing scripts is as @helendduncan suggests above.

Failed to create virtual network gateway on final step.

Basic IP config not supported.

[Link here](https://learn.microsoft.com/en-gb/azure/vpn-gateway/vpn-gateway-vpn-faq#gateways)

```
New-AzVirtualNetworkGateway: Basic IP configuration for VPN Virtual Network Gateways is not supported. Follow the link for more details :
https://go.microsoft.com/fwlink/p/?linkid=2241350
/subscriptions/4aea9c2f-9b6c-42e8-8b09-3594994fe238/resourceGroups/RG_SHM_PROD4_NETWORKING/providers/Microsoft.Network
/virtualNetworkGateways/VNET_SHM_PROD4_GW StatusCode: 400 ReasonPhrase: Bad Request ErrorCode:
PublicIpWithBasicSkuNotAllowedOnVPNGateways ErrorMessage: Basic IP configuration for VPN Virtual Network Gateways is
not supported. Follow the link for more details : https://go.microsoft.com/fwlink/p/?linkid=2241350
/subscriptions/4aea9c2f-9b6c-42e8-8b09-3594994fe238/resourceGroups/RG_SHM_PROD4_NETWORKING/providers/Microsoft.Network
/virtualNetworkGateways/VNET_SHM_PROD4_GW OperationID : e87607a9-8d7c-431c-ab5c-2d4406b9268a
2024-11-19 11:21:01 [FAILURE]: [x] Failed to create virtual network gateway 'VNET_SHM_PROD4_GW'!
Exception: Failed to create virtual network gateway 'VNET_SHM_PROD4_GW'!
```

**Comment by jemrobinson** [comment]
@craddm: I think you fixed the "Basic SKU" for IP addresses issue somewhere else? Is this in the latest v4 release? Can you follow up?

**Comment by helen-d-little** [comment]
its v4.2.2

**Comment by craddm** [comment]
That's odd - it should have been fixed in 4.2.2, as of https://github.com/alan-turing-institute/data-safe-haven/pull/1966

**Comment by helen-d-little** [comment]
Fixed by deleting _GW_PIP as well as per @craddm's initial suggestion
